Content.Shared/Tiles/FloorTileSystem.cs

```diff
@@ -145,6 +147,21 @@ private void OnAfterInteract(EntityUid uid, FloorTileComponent component, AfterI
                     args.Handled = true;
                     return;
                 }
+                if (HasPossibleBaseTurf(currentTileDefinition, baseTurf.ID))
+                {
+                    if (!_tileStack.HasTileStack(tile))
+                    {
+                        _tileStack.CreateTileStack(tile);
+                    }
+                    if (!_stackSystem.Use(uid, 1, stack))
+                        continue;
+                    // current tile gets stored in the tilestack under the new tile
+                    ref var tilestacks = ref Comp<TileStackMapComponent>(gridUid).Data;
+                    tilestacks[tile.GridIndices].Add(tile.GetContentTileDefinition().ID);
+                    PlaceAt(args.User, gridUid, mapGrid, location, currentTileDefinition.TileId, component.PlaceTileSound);
+                    args.Handled = true;
+                    return;
+                }
```

Comment:
Don't add to this over-bloated function with yet another if. This function is now well over 100 lines long and is a nested nightmare.

Reply:
(This isn't your fault but I would show-stopper this PR if I was responsible for merging for this reason alone)